infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Action.java

```diff
@@ -35,14 +35,17 @@
      * Execute the action.
      */
     @Nonnull
-    public void execute(@Nonnull CorfuRuntime runtime) {
-        try {
-            changeStatus(ActionStatus.STARTED);
-            impl(runtime);
-            changeStatus(ActionStatus.COMPLETED);
-        } catch (Exception e) {
-            log.error("execute: error executing action {}", getName(), e);
-            changeStatus(ActionStatus.ERROR);
+    public void execute(@Nonnull CorfuRuntime runtime, int numRetry) {
+        for (int x = 0; x < numRetry; x++) {
```

Comment:
log the retry number and action in case of failures.

Reply:
done.